packages/flutter/lib/src/services/message_codecs.dart

```diff
@@ -209,14 +209,20 @@ class JSONMethodCodec implements MethodCodec {
 ///  * [List]\: `NSArray`
 ///  * [Map]\: `NSDictionary`
 ///
+/// When sending a `java.math.BigInteger` from Java or an `unsigned long`,
+/// `signed long long`, or `unsigned long long` integer from ObjectiveC, it is
+/// converted into a [String] with the hexadecimal representation of the
+/// integer. This codec does not support sending big integers from Dart.
+///
 /// The codec is extensible by subclasses overriding [writeValue] and
 /// [readValueOfType].
 class StandardMessageCodec implements MessageCodec<dynamic> {
   /// Creates a [MessageCodec] using the Flutter standard binary encoding.
   const StandardMessageCodec();
 
-  // The codec serializes messages as outlined below. This format must
-  // match the Android and iOS counterparts.
+  // The codec serializes messages as outlined below. This format must match the
+  // Android and iOS counterparts and cannot change (as it's possible for
+  // someone to end up using this for persistent storage).
```

Comment:
Are we recommending to use it for persistent storage (or are we even ok with it?).
Just thinking that if not we probably shouldn't make any commitment of forward compatibility here...

Reply:
Doesn't matter what commitment we make. People will use it that way anyway.